test/perf/script_generation.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+
+from fairlearn.postprocessing import ThresholdOptimizer
+from fairlearn.reductions import ExponentiatedGradient, GridSearch
+
+
+def generate_script(request, perf_test_configuration, script_name, script_directory):
+    if not os.path.exists(script_directory):
+        os.makedirs(script_directory)
+
+    script_lines = []
+    # imports
+    script_lines.append('from time import time')
```

Comment:
Would it be easier to have a `script.template_py` and do a search/replace on that (even if some of the replaces a multiple lines)?

Reply:
I'm hoping to eventually get to a point where this looks nicer, but I'm scared of "replace". It reads well to have it in order at least. Using replace might mean that we make just a few replacements and nobody understands the logic anymore. I think the next step will have something like:
```
add_imports(script_lines)
add_dataset(script_lines)
add_estimator(script_lines)
add_mitigator(script_lines)
add_analysis(script_lines)

Reply:
I should have added that these will in part come from actual files, so we don't need to write it like this, but there might be a file with all the imports and we can just read it at test time.

Reply:
not doing more than what I did now in this PR though